test/system/irohad_test.cpp

```diff
@@ -224,3 +267,63 @@ TEST_F(IrohadTest, SendQuery) {
         resp.get());
   });
 }
+
+/**
+ * Test verifies that after restarting Iroha with --overwrite-ledger flag there
+ * will be only one block in blockstorage
+ * @given running Iroha with some transactions in storage
+ * @when Iroha is shutdown and restarted with --overwrite-ledger flag
+ * @then blockstorage should have height equal to 1 block and can commit
+ * transactions
+ */
+TEST_F(IrohadTest, RestartWithOverwriteLedger) {
+  launchIroha();
+
+  auto key_manager = iroha::KeysManagerImpl(kAdminId, path_example_);
+  auto key_pair = key_manager.loadKeys();
+  ASSERT_TRUE(key_pair);
+
+  SCOPED_TRACE("From restart with --overwrite-ledger flag test");
+  sendDefaultTxAndCheck(key_pair);
+
+  iroha_process_->terminate();
+
+  launchIroha(config_copy_,
+              path_genesis_.string(),
+              path_keypair_.string(),
+              std::string("--overwrite-ledger"));
+
+  ASSERT_EQ(getBlockCount(), 1);
+
+  SCOPED_TRACE("From restart with --overwrite-ledger flag test");
+  sendDefaultTxAndCheck(key_pair);
+}
+
+/**
+ * Test verifies that after common restart of Iroha it will keep height
```

Comment:
Not clear, what does the word "common" mean. Maybe, "usual" will suit better?
Also, please add some details about height, which should be kept - again, for now it's not crystal clear

Reply:
My suggestion is to omit technical details about blockchain height. 
In my view, it would be better to describe the case from the viewpoint of the ability to accept and serve new transactions.

given an Iroha with some transactions committed ontop of the genesis block
when the Iroha is restarted without --overwrite-ledger flag
then the state is successfully restored, the Iroha accepts and able to commit new transactions